src/source/source_cache.js

```diff
@@ -165,7 +165,7 @@ class SourceCache extends Evented {
     _tileLoaded(tile, id, previousState, err) {
         if (err) {
             tile.state = 'errored';
-            this._source.fire('error', {tile: tile, error: err});
+            if (err.status !== 404) this._source.fire('error', {tile: tile, error: err});
```

Comment:
are there any other cases in which a tile error would 404 for a legitimate reason besides no data in the tile (and thus no tile) ? Should we check the message matches 'Error: not found' ?